### ISSUE TITLE
Import/export tests disregard @needs_xyz decorators

### ISSUE THREAD

**Issue by hannes-ucsc** [BD2KGenomics/toil#894]
The tests insist on testing against Azure as the "other" job store even if Azure isn't supported because the corresponding extra wasn't specified or because the credentials file is missing.

**Comment by hannes-ucsc** [comment]
The following patch fixes that for me:

```diff
Index: src/toil/test/jobStores/jobStoreTest.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/toil/test/jobStores/jobStoreTest.py	(revision 129ffcf84d13dd36820ce14d1bd31d8d21154734)
+++ src/toil/test/jobStores/jobStoreTest.py	(revision )
@@ -370,7 +370,8 @@
             jobStoreTestClasses = [FileJobStoreTest, AWSJobStoreTest, AzureJobStoreTest]
             make_tests(importExportFile,
                        targetClass=cls,
-                       otherJobStore={jsCls.__name__: jsCls for jsCls in jobStoreTestClasses},
+                       otherJobStore={jsCls.__name__: jsCls for jsCls in jobStoreTestClasses 
+                                      if not getattr( jsCls, '__unittest_skip__', False ) },
                        size=dict(zero=0,
                                  one=1,
                                  oneMiB=2**20,
```

**Comment by hannes-ucsc** [comment]
Before applying the patch, think about why it works.

**Comment by hannes-ucsc** [comment]
Will resolves as part of the PR addressing #942.
